File: R programs/MultiSpect_partition.R
MultiSpect_partition <- function(zt,output){
        attach(output)
        nobs = dim(zt)[1]; dimen=dim(zt)[2]
        dev.new()
        hist(nexp_curr[(nwarmup+1):nloop],breaks=0:nexp_max,xlab="Partitions",main="Histogram of Number of Partitions")
        for (j in 1:nexp_max){
                kk <- which(nexp_curr[(nwarmup + 1):nloop] == j)
                if (length(kk) != 0 & j > 1){
                        for (k in 1:(j - 1)){
                                dev.new()
                                plot(xi[[j]][k, kk + nwarmup], type = "l", ylab="Time Point", xlab="Index",
                                     main = paste("Plot of Location of Partition Point", k, 'Given', j, 'Segments'))
                        }
                        for (k in 1:(j - 1)){
                                dev.new()
                                hist(xi[[j]][k, kk + nwarmup],1:nobs,xlab="Time Point",
                                     main=paste("Histogram of Location of Partition Point", k, 'Given', j, 'Segments'))
                        }
                }
        }
        
        prob <- histc(nexp_curr[(nwarmup+1):nloop],1:nexp_max)$cnt/length((nwarmup+1):nloop)
        detach(output)
        return(prob)
}

File: programs/MultiSpect_partition.m
function[post_partitions]=MultiSpect_partition(zt, fit, params) 

dim = size(zt);nobs = dim(1);
post_partitions = (histc(fit(1).nexp_curr(params.nwarmup+1:params.nloop),1:params.nexp_max)'/length(params.nwarmup+1:params.nloop))';

figure
histogram(fit(1).nexp_curr(params.nwarmup+1:params.nloop),'FaceAlpha',1,...
    'Normalization','probability','BinMethod','integers','BinWidth',.5,'BinLimits',[1,10])
title('Histogram of the number of partitions')
xlabel('The number of partitions')
ylabel('Probability')
for j=1:params.nexp_max
	kk=find(fit(1).nexp_curr(params.nwarmup+1:params.nloop)==j);
    if ~isempty(kk) && j>1
        figure
        hold
        title(['Plot of Partition Points with total number of partitions equal to ',int2str(j-1)])
        for k=1:j-1
            plot(fit(j).xi(k,kk+params.nwarmup))
        end
        for k=1:j-1
            figure
            hold
            title(['Histogram for location of partition ', int2str(k), ',' ' with total number of segment equal to ',int2str(j)])
            histogram(fit(j).xi(k,kk+params.nwarmup),'FaceAlpha',1,...
                'Normalization','probability','BinMethod','integers','BinWidth',20,'BinLimits',[1,nobs])
        end
    end
end

end
